BUG: `openmp/cmake/HandleOpenMPOptions.cmake` uses CMake function `append` without defining it

While [building](https://github.com/conda-forge/openmp-feedstock/pull/113) openmp 18.1.0-rc1 on windows, I get:
```
CMake Error at cmake/HandleOpenMPOptions.cmake:47 (append):
  Unknown CMake command "append".
```

This is due to https://github.com/llvm/llvm-project/blame/release/18.x/openmp/cmake/HandleOpenMPOptions.cmake using `append` since https://github.com/llvm/llvm-project/commit/15fdc7646ca24506ccb6a59861da8f3b52ab031c, but not defining it -- in contrast to `append_if` which explicitly duplicates the definition from [`HandleLLVMOptions.cmake`](https://github.com/llvm/llvm-project/blob/release/18.x/llvm/cmake/modules/HandleLLVMOptions.cmake):
https://github.com/llvm/llvm-project/blob/400a02bb281ba0de79211ab53d931e44dd76733f/openmp/cmake/HandleOpenMPOptions.cmake#L1-L10

I don't know why this file does not (or cannot) do `include(HandleLLVMOptions)`, but if that's not an option, then it needs to similarly vendor the definition of `append`.

CC @aganea

--- a/openmp/cmake/HandleOpenMPOptions.cmake
+++ b/openmp/cmake/HandleOpenMPOptions.cmake
@@ -52,9 +52,9 @@ append_if(OPENMP_HAVE_DATA_SECTIONS "-fdata-sections" CMAKE_C_FLAGS CMAKE_CXX_FL
 
 if (MSVC)
   # Disable "warning C4201: nonstandard extension used: nameless struct/union"
-  append("-wd4201" CMAKE_C_FLAGS CMAKE_CXX_FLAGS )
+  append("-wd4201" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
   
   # Disable "warning C4190: '__kmpc_atomic_cmplx8_rd' has C-linkage specified, but returns
   # UDT '__kmp_cmplx64_t' which is incompatible with C"
-  append("-wd4190" CMAKE_C_FLAGS CMAKE_CXX_FLAGS )
+  append("-wd4190" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
 endif()

--- a/openmp/cmake/HandleOpenMPOptions.cmake
+++ b/openmp/cmake/HandleOpenMPOptions.cmake
@@ -44,9 +44,9 @@ append_if(OPENMP_HAVE_DATA_SECTIONS "-fdata-sections" CMAKE_C_FLAGS CMAKE_CXX_FL
 
 if (MSVC)
   # Disable "warning C4201: nonstandard extension used: nameless struct/union"
-  append("-wd4201" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
+  append_if(TRUE "-wd4201" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
   
   # Disable "warning C4190: '__kmpc_atomic_cmplx8_rd' has C-linkage specified, but returns
   # UDT '__kmp_cmplx64_t' which is incompatible with C"
-  append("-wd4190" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
+  append_if(TRUE "-wd4190" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
 endif()

--- a/openmp/cmake/HandleOpenMPOptions.cmake
+++ b/openmp/cmake/HandleOpenMPOptions.cmake
@@ -9,6 +9,14 @@ if (NOT COMMAND append_if)
   endfunction()
 endif()
 
+if (NOT COMMAND append)
+  function(append value)
+    foreach(variable ${ARGN})
+      set(${variable} "${${variable}} ${value}" PARENT_SCOPE)
+    endforeach(variable)
+  endfunction()
+endif()
+
 # MSVC and clang-cl in compatibility mode map -Wall to -Weverything.
 # TODO: LLVM adds /W4 instead, check if that works for the OpenMP runtimes.
 if (NOT MSVC)
@@ -44,9 +52,9 @@ append_if(OPENMP_HAVE_DATA_SECTIONS "-fdata-sections" CMAKE_C_FLAGS CMAKE_CXX_FL
 
 if (MSVC)
   # Disable "warning C4201: nonstandard extension used: nameless struct/union"
-  append_if(TRUE "-wd4201" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
+  append("-wd4201" CMAKE_C_FLAGS CMAKE_CXX_FLAGS )
   
   # Disable "warning C4190: '__kmpc_atomic_cmplx8_rd' has C-linkage specified, but returns
   # UDT '__kmp_cmplx64_t' which is incompatible with C"
-  append_if(TRUE "-wd4190" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
+  append("-wd4190" CMAKE_C_FLAGS CMAKE_CXX_FLAGS )
 endif()

